stellar/deploy-contract.js

```diff
@@ -96,36 +101,83 @@ async function processCommand(options, config, chain) {
     await broadcast(operation, wallet, chain, 'Initialized contract', options);
 }
 
+async function processUpgrade(options, _, chain) {
+    const { contractName, contractId, privateKey, wasmPath, newWasmHash } = options;
+    const { rpc, networkType } = chain;
+    const networkPassphrase = getNetworkPassphrase(networkType);
+
+    if (!chain.contracts) {
+        chain.contracts = {};
+    }
+
+    const args = `--source ${privateKey} --rpc-url ${rpc} --network-passphrase "${networkPassphrase}"`;
+
+    let cmd;
+
+    if (options.install) {
+        cmd = `${stellarCmd} contract install --wasm ${wasmPath} ${args}`;
+    } else {
+        cmd = `${stellarCmd} contract invoke --id ${contractId} ${args} -- upgrade --new_wasm_hash ${newWasmHash}`;
+    }
+
+    printInfo('Executing command', cmd);
+    printInfo('Deploying contract', contractName);
+
+    const result = execSync(cmd, { encoding: 'utf-8', stdio: 'pipe' }).trimEnd();
+
+    if (options.install) {
+        printInfo('Contract WASM hash', result);
+    } else {
+        printInfo('Upgraded contract successfully!');
+    }
+}
+
 async function mainProcessor(options, processor) {
     const config = loadConfig(options.env);
     await processor(options, config, getChainConfig(config, options.chainName));
     saveConfig(config, options.env);
 }
 
 function main() {
-    const program = new Command();
-    program.name('deploy-contract').description('Deploy Axelar Soroban contracts on Stellar');
-
-    addBaseOptions(program, { address: true });
-
-    program.addOption(new Option('--initialize', 'initialize the contract'));
-    program.addOption(new Option('--contract-name <contractName>', 'contract name to deploy').makeOptionMandatory(true));
-    program.addOption(new Option('--wasm-path <wasmPath>', 'path to the WASM file').makeOptionMandatory(true));
-    program.addOption(new Option('--nonce <nonce>', 'optional nonce for the signer set'));
-    program.addOption(
-        new Option('--previous-signers-retention <previousSignersRetention>', 'previous signer retention').default(15).argParser(Number),
-    );
-
-    program.addOption(
-        new Option(
-            '--domain-separator <domainSeparator>',
-            'domain separator (pass in the keccak256 hash value OR "offline" meaning that its computed locally)',
-        ).default('offline'),
-    );
-
-    program.action((options) => {
-        mainProcessor(options, processCommand);
-    });
+    // 1st level command
+    const program = new Command('deploy-contract').description('Deploy/Upgrade Soroban contracts on Stellar');
+
+    // 2nd level deploy command
+    const deployCmd = new Command('deploy')
```

Comment:
```suggestion
    const deployCmd = new Command('deploy <contract-name>')
```
make it a positional arg since it's required

Reply:
good catch, updated.